projects/addon-doc/components/documentation/directives/documentation-property-connector.directive.ts

```diff
@@ -139,15 +167,68 @@ export class TuiDocDocumentationPropertyConnectorDirective<T>
             isValueAvailableByKey && this.documentationPropertyValues
                 ? this.documentationPropertyValues.indexOf(value as T)
                 : value;
+        const computedDefaultValue =
+            isValueAvailableByKey &&
+            this.documentationPropertyValues &&
+            this.documentationPropertyDefaultValue !== undefined
+                ? this.documentationPropertyValues.indexOf(
+                      this.documentationPropertyDefaultValue,
+                  )
+                : this.documentationPropertyDefaultValue;
 
         const suffix = isValueAvailableByKey ? SERIALIZED_SUFFIX : '';
         const propName = this.documentationPropertyName + suffix;
 
-        tree.queryParams = {
-            ...tree.queryParams,
-            [propName]: computedValue,
-        };
+        if (computedDefaultValue === computedValue) {
+            delete tree.queryParams[propName];
+        } else {
+            tree.queryParams[propName] = computedValue;
+        }
```

Comment:
why not immutable structure?

Reply:
I don't understand why... the tree is created only for this function and is used only in it.